arch/arm64/common/head.S

```diff
@@ -399,6 +341,12 @@ ENTRY(__create_page_tables)
 	mov	x3, x24				// PHYS_OFFSET
 	create_block_map x0, x7, x3, x5, x6
 
+#ifdef CONFIG_GEM5
+.global simpoint_gen_pte_ret
+	b simpoint_gen_pte
+simpoint_gen_pte_ret:
```

Comment:
Can this work:
        ldr     x27, =simpoint_gen_pte_done
simpoint_gen_pte_gen:

At the end of simpoint_gen_pte:
        br      x27

Reply:
Also need to add comments for why you need to do 'ret' in this way.

Reply:
Final label name: simpoint_heap_map_entry/exit
Comment added.